server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -269,6 +269,17 @@ public static IndexMergePolicy fromString(String text) {
         Property.IndexScope
     );
 
+    /**
+     * Index setting describing the maximum number of nested scopes in queries and documents.
+     * The default maximum of 10. 0 means no nesting allowed.
+     */
+    public static final Setting<Integer> MAX_NESTED_DEPTH_SETTING = Setting.intSetting(
```

Comment:
May be let's add `QUERY` to the name to emphasize this is query specific?

```suggestion
    public static final Setting<Integer> MAX_QUERY_NESTED_DEPTH_SETTING = Setting.intSetting(
```